hdf5.spec

```diff
@@ -391,6 +386,12 @@ rm %{buildroot}%{_mandir}/man1/h5p[cf]c*.1
 mkdir -p %{buildroot}%{_libdir}/%{name}
 mv %{buildroot}%{_libdir}/libhdf5_java.so %{buildroot}%{_libdir}/%{name}/
 
+# install the hdf5.jar files; SLES is not doing the install
```

Comment:
So this gets installed for EL7 but not when run on Leap 15?  If so:

- is this a bug upstream?
- can we identify in the source why this doesn't happen on Leap 15 and patch it there rather than hacking it in here?

Reply:
The code that I added for SLES is what I has seen in the EL7 log.   EL7 install the jar files in /usr/lib64

adding: hdf/hdf5lib/HDF5Constants.class(in = 41921) (out= 19167)(deflated 54%)
adding: hdf/hdf5lib/HDFNativeData.class(in = 3191) (out= 1569)(deflated 50%)
adding: hdf/hdf5lib/HDF5GroupInfo.class(in = 2077) (out= 1081)(deflated 47%)
 /usr/bin/mkdir -p '/builddir/build/BUILDROOT/hdf5-1.12.0-1.g4aa4036562.el7.x86_64/usr/lib64'
 /usr/bin/install -c -m 644 hdf5.jar '/builddir/build/BUILDROOT/hdf5-1.12.0-1.g4aa4036562.el7.x86_64/usr/lib64'
 /usr/bin/mkdir -p '/builddir/build/BUILDROOT/hdf5-1.12.0-1.g4aa4036562.el7.x86_64/usr/lib64'

In SLES this install step was not happening and the job was failing because it could not find the hdf5.jar file in BUILDROOT so I added it to the spec file for SLES

Reply:
This was the Leap rpm failure I was getting that led me to install hdf5.jar in BUILDROOT....../usr/lib64
Processing files: java-hdf5-1.12.0-1.g4aa4036562.suse.lp151.x86_64
error: File not found: /builddir/build/BUILDROOT/hdf5-1.12.0-1.g4aa4036562.suse.lp151.x86_64/usr/lib64/hdf5.jar
RPM build errors:
    File not found: /builddir/build/BUILDROOT/hdf5-1.12.0-1.g4aa4036562.suse.lp151.x86_64/usr/lib64/hdf5.jar
Child return code was: 1
EXCEPTION: [Error()]

I had changed it to conditionalize for sles to use %{_jnidir} and this did not make a difference...
Processing files: java-hdf5-1.12.0-1.g4aa4036562.suse.lp151.x86_64
error: File not found: /builddir/build/BUILDROOT/hdf5-1.12.0-1.g4aa4036562.suse.lp151.x86_64/usr/lib64/java/hdf5.jar
RPM build errors:
    File not found: /builddir/build/BUILDROOT/hdf5-1.12.0-1.g4aa4036562.suse.lp151.x86_64/usr/lib64/java/hdf5.jar

Reply:
> In SLES this install step was not happening and the job was failing because it could not find the hdf5.jar file in BUILDROOT so I added it to the spec file for SLES

So this is the root issue that we should figure out rather than hacking it in here.